### ISSUE TITLE
RollingDestination trying to roll stream destinations outside of a user context

### ISSUE THREAD

**Issue by stroomdev66** [gchq/stroom#2906]
The job that periodically rolls rolling stream destinations does so outside of any user context. This is causing FsTarget.close to error when the system tries to update the status of a stream. To fix this RolliingDestination.tryFlushAndRoll() should be run as the processing user.

**Comment by at055612** [comment]
Fixed in > 7.0-beta.209

**Comment by at055612** [comment]
Still seeing errors in stroom.pipeline.destination.RollingDestination(,s) - No user is currently logged in...

RollingDestination.java:244

**Comment by stroomdev10** [comment]
[rolling.txt](https://github.com/gchq/stroom/files/8974506/rolling.txt)

**Comment by at055612** [comment]
In-lining stack
```
No user is currently logged in 
java.io.IOException: No user is currently logged in
	at stroom.pipeline.destination.RollingDestination.lambda$throwFirstAsIOExceptions$0(RollingDestination.java:244)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at stroom.pipeline.destination.RollingDestination.throwFirstAsIOExceptions(RollingDestination.java:246)
	at stroom.pipeline.destination.RollingDestination.tryFlushAndRoll(RollingDestination.java:157)
	at stroom.pipeline.destination.RollingDestinations.lambda$rollAll$3(RollingDestinations.java:168)
	at java.base/java.util.concurrent.ConcurrentHashMap$ForEachMappingTask.compute(ConcurrentHashMap.java:5081)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: stroom.security.impl.exception.AuthenticationException: No user is currently logged in
	at stroom.security.impl.SecurityContextImpl.hasAppPermission(SecurityContextImpl.java:122)
	at stroom.security.impl.SecurityContextImpl.isAdmin(SecurityContextImpl.java:75)
	at stroom.feed.impl.MetaSecurityFilterImpl.getExpression(MetaSecurityFilterImpl.java:48)
	at stroom.meta.impl.MetaServiceImpl.lambda$addPermissionConstraints$16(MetaServiceImpl.java:647)
	at java.base/java.util.Optional.map(Optional.java:258)
	at stroom.meta.impl.MetaServiceImpl.addPermissionConstraints(MetaServiceImpl.java:645)
	at stroom.meta.impl.MetaServiceImpl.updateStatus(MetaServiceImpl.java:177)
	at stroom.meta.impl.MetaServiceImpl.updateStatus(MetaServiceImpl.java:141)
	at stroom.data.store.impl.fs.FsTarget.unlock(FsTarget.java:272)
	at stroom.data.store.impl.fs.FsTarget.close(FsTarget.java:246)
	at stroom.pipeline.destination.RollingStreamDestination.afterRoll(RollingStreamDestination.java:83)
	at stroom.pipeline.destination.RollingDestination.roll(RollingDestination.java:197)
	at stroom.pipeline.destination.RollingDestination.tryFlushAndRoll(RollingDestination.java:147)
	... 8 common frames omitted
	
	
	No user is currently logged in 
stroom.security.impl.exception.AuthenticationException: No user is currently logged in
	at stroom.security.impl.SecurityContextImpl.hasAppPermission(SecurityContextImpl.java:122)
	at stroom.security.impl.SecurityContextImpl.isAdmin(SecurityContextImpl.java:75)
	at stroom.feed.impl.MetaSecurityFilterImpl.getExpression(MetaSecurityFilterImpl.java:48)
	at stroom.meta.impl.MetaServiceImpl.lambda$addPermissionConstraints$16(MetaServiceImpl.java:648)
	at java.base/java.util.Optional.map(Optional.java:258)
	at stroom.meta.impl.MetaServiceImpl.addPermissionConstraints(MetaServiceImpl.java:645)
	at stroom.meta.impl.MetaServiceImpl.updateStatus(MetaServiceImpl.java:177)
	at stroom.meta.impl.MetaServiceImpl.updateStatus(MetaServiceImpl.java:141)
	at stroom.data.store.impl.fs.FsTarget.unlock(FsTarget.java:272)
	at stroom.data.store.impl.fs.FsTarget.close(FsTarget.java:246)
	at stroom.pipeline.destination.RollingStreamDestination.afterRoll(RollingStreamDestination.java:83)
	at stroom.pipeline.destination.RollingDestination.roll(RollingDestination.java:197)
	at stroom.pipeline.destination.RollingDestination.tryFlushAndRoll(RollingDestination.java:147)
	at stroom.pipeline.destination.RollingDestinations.lambda$rollAll$3(RollingDestinations.java:168)
	at java.base/java.util.concurrent.ConcurrentHashMap$ForEachMappingTask.compute(ConcurrentHashMap.java:5081)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by at055612** [comment]
Fixed in > 7.0-beta.210
